math_unit.pas

```diff
@@ -25303,11 +25318,23 @@ procedure drawtimbcl(retcurve: boolean);       // mark timber centre-line
     // shift required to rotation point, to suit the curving..
     y_curmod := y_curtimb - yeq;
 
-    if midline = False then
-      calc_fill_timber_mark(EMarkCode(4 + shove_this))  // 4 = centre-line,  44 = selected timber.
+    if midline then
+       case timberStatus of
+         eTS_Normal:
+           code := eMC_14_TimberCLMidline;
+         eTS_Shoved:
+           code := eMC_54_ShovingTimberCLMidline;
+         eTS_Selected:
+           code := eMC_54_ShovingTimberCLMidline;
+       end
     else
-      calc_fill_timber_mark(EMarkCode(14 + shove_this));
-    // 14,54 = timber centre-line, use solid line if drawing rail centrelines. (for rivets?)
+       case timberStatus of
+         eTS_Normal: code := eMC_4_TimberCL;
+         eTS_Shoved: code := eMC_44_ShovingTimberCL;
+         eTS_Selected: code := eMC_44_ShovingTimberCL;
+     end;
+    calc_fill_timber_mark(code);
+
```

Comment:
What is here is strictly correct, but not how I would (normally) arrange things. I prefer to always use begin/end blocks even when not strictly necessary:

```
if midline then begin
  case...
  end;
end
else begin
  case...
  end;
end;
```